packages/data-viz/src/core/HeatmapChart/hooks/utils.ts

```diff
@@ -82,94 +137,248 @@ export interface CreateChartOptionsProps {
   grid?:
     | EChartsOption["grid"]
     | ((defaultOption: EChartsOption["grid"]) => EChartsOption["grid"]);
+  /**
+   * The options object to be passed to echarts.setOption()
+   * https://echarts.apache.org/en/option.html
+   */
+  options?: EChartsOption;
+  /**
+   * Event listeners for the chart
+   * https://echarts.apache.org/en/api.html#events
+   */
+  onEvents?: Record<string, (event: unknown, chart: ECharts) => void>;
 }
 
-export function createChartOptions({
-  data,
-  xAxisData,
-  yAxisData,
-  width,
-  height,
-  encode,
-  itemStyle = DEFAULT_ITEM_STYLE,
-  symbolSize,
-  grid: gridProp,
-}: CreateChartOptionsProps): EChartsOption {
-  const defaultGrid = {
-    height: `${height}px`,
-    left: 0,
-    top: 0,
-    // (atarashansky): this is the key change to align x and y axis
-    // labels to fixed spacings
-    width: `${width}px`,
-  };
+export function createChartOptions(
+  props: CreateChartOptionsProps
+): EChartsOption {
+  const {
+    axisPointer,
+    camera,
+    data,
+    dataZoom,
+    emphasis,
+    encode,
+    grid: gridProp,
+    itemStyle = DEFAULT_ITEM_STYLE,
+    options,
+    symbolSize,
+    symbol = "rect",
+  } = props;
+
+  const {
+    defaultAxisPointer,
+    defaultDataZoom,
+    defaultEmphasis,
+    defaultGrid,
+    defaultXAxis,
+    defaultYAxis,
+  } = generateDefaultValues(props);
 
   const customGrid =
     typeof gridProp === "function" ? gridProp(defaultGrid) : gridProp;
 
+  const {
+    axisPointer: optionsAxisPointer,
+    dataZoom: optionsDataZoom,
+    series: optionsSeries,
+    xAxis: optionsXAxis,
+    yAxis: optionsYAxis,
+    ...optionsRest
+  } = options || {};
+
   return {
     animation: false,
-    /**
-     * Display reference line and axis value under mouse pointer
-     * https://echarts.apache.org/en/option.html#axisPointer
-     */
-    axisPointer: {
-      label: { show: false },
-      show: true,
-      triggerOn: "mousemove",
-    },
+    axisPointer: Object.assign(
+      defaultAxisPointer,
+      axisPointer,
```

Comment:
seems like `axisPointer` technically could be an array, so that might mess up with `Object.assign()`? Do we need the same `Array.isArray ?` check as other options do?

Reply:
Oh, good point! I'll fix it. 😯